Psychtoolbox/PsychDemos/PsychTutorials/AlphaImageTutorial.m

```diff
@@ -40,202 +40,213 @@
-    Screen(w,'BlendFunction',GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
-
-    % Find the color values which correspond to white and black.
-    black=BlackIndex(screenNumber);
-    gray=GrayIndex(screenNumber); 
-
-    % Import image and and convert it, stored in
-    % MATLAB matrix, into a Psychtoolbox OpenGL texture using 'MakeTexture';
-    myimgfile = [ PsychtoolboxRoot 'PsychDemos' filesep 'AlphaImageDemo' filesep 'konijntjes1024x768.jpg'];
-    fprintf('Using image ''%s''\n', myimgfile);
-    imdata=imread(myimgfile);
-    
-    % Crop image if it is larger then screen size. There's no image scaling
-    % in maketexture
-    [iy, ix, iz]=size(imdata); %#ok<NASGU>
-    [wW, wH]=WindowSize(w);
-    if ix>wW || iy>wH
-        fprintf('Image size exceeds screen size\n');
-        fprintf('Image will be cropped\n');
-    end
-    if ix>wW
-        cl=round((ix-wW)/2);
-        cr=(ix-wW)-cl;
-    else
-        cl=0;
-        cr=0;
-    end
-    if iy>wH
-        ct=round((iy-wH)/2);
-        cb=(iy-wH)-ct;
-    else
-        ct=0;
-        cb=0;
-    end
-
-    % Set up texture and rects
-    imagetex=Screen('MakeTexture', w, imdata(1+ct:iy-cb, 1+cl:ix-cr,:));
-    tRect=Screen('Rect', imagetex);
-    [ctRect, dx, dy]=CenterRect(tRect, wRect);
-
-    % We create a Luminance+Alpha matrix for use as transparency mask:
-    % Layer 1 (Luminance) is filled with luminance value 'gray' of the
-    % background.
-    ms=100;
-    transLayer=2;
-    [x,y]=meshgrid(-ms:ms, -ms:ms);
-    maskblob=uint8(ones(2*ms+1, 2*ms+1, transLayer) * gray);
-    size(maskblob);
-    
-    % Layer 2 (Transparency aka Alpha) is filled with gaussian transparency
-    % mask.
-    xsd=ms/2.0;
-    ysd=ms/2.0;
-    maskblob(:,:,transLayer)=uint8(round(255 - exp(-((x/xsd).^2)-((y/ysd).^2))*255));
-    
-    % Build a single transparency mask texture
-    masktex=Screen('MakeTexture', w, maskblob);
-    mRect=Screen('Rect', masktex);
-    fprintf('Size image texture: %d x %d\n', RectWidth(tRect), RectHeight(tRect));
-    fprintf('Size  mask texture: %d x %d\n', RectWidth(mRect), RectHeight(mRect));
-
-    % Blank sceen
-    Screen('FillRect',w, gray);
-    Screen('Flip', w);
-
-    % initialize mouse position at center
-    [a,b]=WindowCenter(w);
-    SetMouse(a,b,screenNumber); 
-    HideCursor;
-
-    % Bump priority for speed        
+	% Say hello in command window
+	fprintf('\nAlphaImageDemo\n');
+	
+	% Get the list of screens and choose the one with the highest screen number.
+	% Screen 0 is, by definition, the display with the menu bar. Often when
+	% two monitors are connected the one without the menu bar is used as
+	% the stimulus display.  Chosing the display with the highest dislay number is
+	% a best guess about where you want the stimulus displayed.
+	screenNumber=max(Screen('Screens'));
+	
+	% Open a double buffered fullscreen window and draw a gray background
+	% and front and back buffers.
+	PsychDefaultSetup(2);
```

Comment:
PsychDefaultSetup(2) should actually replace the AssertOpenGL at the top, as it should go to the top and includes AssertOpenGL;